src/main/modules/i18n/index.ts

```diff
@@ -1,18 +1,28 @@
 import { Express } from 'express';
+import Negotiator from 'negotiator';
+
+export enum SupportedLanguages {
+  En = 'en',
+  Cy = 'cy',
+}
 
 /**
  * Module that enables toggling between languages
  */
 export class LanguageToggle {
-  static supportedLanguages = ['en', 'cy'];
+  static supportedLanguages = [SupportedLanguages.En, SupportedLanguages.Cy];
 
   public enableFor(app: Express): void {
     app.use((req, res, next) => {
-      if (req.method === 'GET' && req.query['lng']) {
-        const requestedLanguage = req.query['lng'] as string;
-
+      if (req.method === 'GET' && (req.query['lng'] || !req.session['lang'])) {
+        // User selected language
+        const requestedLanguage = req.query['lng'] as SupportedLanguages;
```

Comment:
would this not cause an issue if req.query['lng'] returned undefined, and req.session['lang'] wasn't set?

Reply:
I'm assuming SupportedLanguages type can be undefined, in which case it would work. But why do we need to add the check of` !req.session['lang']` ? I can't see the scenario this makes a difference for

Reply:
the only difference I can see this making is if req.query['lng'] is undefined and so is the req.session['lang'], but all this does would be to set the req.session['lang'] to undefined again?